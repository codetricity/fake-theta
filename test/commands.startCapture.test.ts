import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('POST /osc/commands/execute startCapture', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.startCapture',
      })
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });
});

describe('THETA X POST /osc/commands/execute startCapture', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.startCapture',
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });
});

describe('THETA Z1 POST /osc/commands/execute startCapture', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.startCapture',
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });
});

describe('Unknown Type POST /osc/commands/execute startCapture', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera.startCapture',
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
