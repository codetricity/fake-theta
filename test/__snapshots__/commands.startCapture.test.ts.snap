// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute startCapture should respond with results 1`] = `
{
  "id": 10,
  "name": "camera.startCapture",
  "state": "done",
}
`;

exports[`THETA X POST /osc/commands/execute startCapture should respond with results 1`] = `
{
  "id": 10,
  "name": "camera.startCapture",
  "state": "done",
}
`;

exports[`THETA Z1 POST /osc/commands/execute startCapture should respond with results 1`] = `
{
  "name": "camera.startCapture",
  "state": "done",
}
`;
